stock_available_to_promise_release/models/stock_move.py

```diff
@@ -603,6 +614,79 @@ def _get_chained_moves_iterator(self, chain_field):
+            if not pickings:
+                continue
+            return_type = picking_type.return_picking_type_id
+            wiz_values = {
+                "picking_id": fields.first(pickings).id,
+                "location_id": return_type.default_location_dest_id.id,
+            }
+            product_return_moves = []
+            if not return_type:
+                message = _(
+                    "The operation {picking_names} is done and cannot be returned",
+                    picking_names=", ".join(pickings.mapped("name")),
+                )
+                raise UserError(message)
+            for move in moves:
+                # Cannot return an unprocessed move
+                if move.state != "done":
+                    continue
+                product = move.product_id
+                uom = product.uom_id
+                qty_to_return = qty_to_return_per_move.get(move.id, 0)
+                # Cannot return 0 qty
+                if float_is_zero(qty_to_return, precision_rounding=uom.rounding):
+                    continue
+                return_move_vals = {
+                    "product_id": product.id,
+                    "quantity": qty_to_return,
+                    "uom_id": uom.id,
+                    "move_id": move.id,
+                }
+                product_return_moves.append((0, 0, return_move_vals))
+            if product_return_moves:
+                wiz_values["product_return_moves"] = product_return_moves
+                return_wiz = self.env["stock.return.picking"].create(wiz_values)
+                return_wiz.create_returns()
+        return True
+
+    def _get_returneable_qty_per_move(self, qty_to_return, qty_to_return_per_move):
+        returneable_qty = 0
+        for move in self:
+            rounding = move.product_id.uom_id.rounding
+            # As a move might have multiple dest ids, we might have
+            # already planned to return a few units already.
+            # Get it, and deduce it from the returneable qty
+            move_qty_planned = qty_to_return_per_move.get(move.id, 0)
+            # A move might already have return moves linked to it, deduce their quantity
+            move_returned_qty = sum(
+                move.returned_move_ids.filtered(lambda m: m.state != "cancel").mapped(
+                    "product_qty"
+                )
+            )
+            move_returneable_qty = min(
+                qty_to_return, move.product_qty - move_returned_qty - move_qty_planned
+            )
+            if float_is_zero(move_returneable_qty, precision_rounding=rounding):
+                continue
+            # Update the quantity
+            qty_to_return_per_move[move.id] += move_returneable_qty
+            qty_to_return -= move_returneable_qty
+            returneable_qty += move_returneable_qty
+            if float_is_zero(qty_to_return, precision_rounding=rounding):
+                break
+        return returneable_qty
```

Comment:
We have to return the complete `qty_to_return` otherwise there is an issue
```suggestion
            if float_is_zero(qty_to_return, precision_rounding=rounding):
                break
        else:
             raise UserError("Unrelease cannot return the required quantity")
```

Reply:
it's maybe not a good idea my proposal in case someone cancel the pack before the ship